tensorflow/python/ops/metrics_impl.py

```diff
@@ -258,8 +258,11 @@ def _streaming_confusion_matrix(labels, predictions, num_classes, weights=None):
     total_cm: A `Tensor` representing the confusion matrix.
     update_op: An operation that increments the confusion matrix.
   """
+  # TODO: This should be int64 when weights is None, but legacy bug-
+  # compatibility requires keeping this as float64 in that case.
+  cm_dtype = dtypes.float64
```

Comment:
The more straightforward fix would be to just invert the branches.

However, if anybody is importing this function from here, and is e.g. dividing this by whatever in Python 2 without doing a float cast, this would then change the numerical results that they see.

Given that this bug has been around for a while, I think it'd perhaps be justifiable to keep the old behavior for cases where weights aren't specified.

I personally wouldn't be affected by a correct fix of just inverting the branches, though, and would be happy to amend this PR if such a fix would be better.

Reply:
I think inverting the branches is better here.

Reply:
Sure, will do.